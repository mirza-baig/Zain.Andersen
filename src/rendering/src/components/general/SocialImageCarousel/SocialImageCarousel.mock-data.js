export const stackData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'SocialImageCarousel',
  },
  fields: {
    genericFilter: {
      value: 'productId:ES-CAS',
    },
    siteName: {
      value: 'andersenwindows-6o5qt0',
    },
    slideFormat: {
      id: 'a519844c-58eb-41c2-ae98-b9eb94eb7cf6',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/carousel-inline-stack/4-image',
      name: '4 Image',
      displayName: '4 Image',
      fields: {
        Value: {
          value: 'stack',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    containerID: {
      value: 'homepage',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/headline-levels/heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'Real projects. Real perspectives.',
    },
    body: {
      value: 'This body text only shows up if there are 2 slides selected.',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export const inlineData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'SocialImageCarousel',
  },
  fields: {
    genericFilter: {
      value: 'productId:ES-CAS',
    },
    siteName: {
      value: 'andersenwindows-6o5qt0',
    },
    slideFormat: {
      id: '3eb18869-c0b0-4352-bf58-08d205bd2501',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/carousel-inline-stack/2-image',
      name: '2 Image',
      displayName: '2 Image',
      fields: {
        Value: {
          value: 'inline',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    containerID: {
      value: 'homepage',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/headline-levels/heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'Social Image Carousel Inline',
    },
    body: {
      value: 'This body text only shows up if there are 2 slides selected.',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};
