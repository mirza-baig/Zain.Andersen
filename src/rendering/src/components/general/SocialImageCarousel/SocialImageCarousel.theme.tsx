// Lib
import classNames from 'classnames';
import { ThemeFile } from 'lib/context/ThemeContext';

const headingRuleLine =
  'before:block before:w-[calc(100%_-_40px)] before:ml-auto before:border-t before:mb-xs before:border-gray';

export const SocialImageCarouselTheme: ThemeFile = {
  aw: {
    classes: {
      wrapper: 'flex flex-col md:flex-row col-span-2 md:col-span-6 items-center theme-white',
      wrapper2: 'col-span-12',
      headlineClass: 'p-3 max-w-[315px] md:min-w-[350px]',
      headlineText: 'font-sans font-heavy text-m md:w-[350px] mb-3',
      headlineText2: 'font-sans font-heavy text-m mb-3 ml-10',
      bodyCopyClass: ' text-body text-dark-gray',
      carouselInline:
        'w-[375px] md:w-[275px] mmd:w-[403px] ml:w-[611px] mml:w-[627px] lg:w-[845px]',
      carouselStack: 'w-full',
    },
    defaultProductFilter: 'ES-CAS',
    defaultSiteName: 'andersenwindows-6o5qt0',
  },
  rba: {
    classes: {
      wrapper: classNames(
        'flex flex-col md:flex-row col-span-2 md:col-span-6 items-center theme-white'
      ),
      wrapper2: classNames('col-span-12', headingRuleLine),
      headlineClass: 'p-3 max-w-[315px] md:min-w-[350px]',
      headlineText: 'font-sans font-bold text-s md:w-[350px] mb-3',
      headlineText2: '!font-serif font-bold text-sm-xs md:text-xxs mb-3 ml-10',
      bodyCopyClass: ' text-body text-dark-gray',
      carouselInline: 'w-[375px] lg:w-[845px] md:w-[575px]',
      carouselStack: 'w-full',
    },
    defaultProductFilter: 'doublehung',
    defaultSiteName: 'renewalbyandersen-sb1oge',
  },
};
