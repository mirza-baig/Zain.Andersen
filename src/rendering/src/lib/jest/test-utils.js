// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
