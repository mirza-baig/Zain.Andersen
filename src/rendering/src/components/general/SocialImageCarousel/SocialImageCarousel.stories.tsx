// Global
import { Story, Meta } from '@storybook/react';
// Local
import SocialImageCarousel, { SocialImageCarouselProps } from './SocialImageCarousel';
import { inlineData, stackData } from './SocialImageCarousel.mock-data';

export default {
  title: 'General/SocialImageCarousel',
  component: SocialImageCarousel,
} as Meta;

const Template: Story<SocialImageCarouselProps> = (props) => <SocialImageCarousel {...props} />;

export const Inline = Template.bind({});
Inline.args = inlineData;

export const Stack = Template.bind({});
Stack.args = stackData;
