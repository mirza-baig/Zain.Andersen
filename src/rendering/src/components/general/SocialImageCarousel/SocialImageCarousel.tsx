// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { SocialImageCarouselTheme } from './SocialImageCarousel.theme';
import Script from 'next/script';
import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';

export type SocialImageCarouselProps =
  Feature.EnterpriseWeb.Enterprise.Components.Social.SocialImageCarousel;
const SocialImageCarousel = (props: SocialImageCarouselProps) => {
  const { themeData } = useTheme(SocialImageCarouselTheme);
  const { themeName } = useTheme();

  const siteName =
    props.fields?.siteName.value ?? themeName === 'aw'
      ? 'andersenwindows-6o5qt0'
      : 'renewalbyandersen-sb1oge';

  const containerId = props.fields?.containerID?.value;
  const genericFilter = props.fields?.genericFilter?.value;
  const slideFormat = props.fields?.slideFormat?.fields?.Value?.value ?? 'inline';

  return (
    <Component variant="lg" dataComponent="general/socialimagecarousel" {...props}>
      <div
        className={
          slideFormat === 'inline' ? themeData.classes.wrapper : themeData.classes.wrapper2
        }
      >
        {slideFormat === 'inline' ? (
          <div className={themeData.classes.headlineClass}>
            <Headline {...props} />
            <BodyCopy classes={themeData.classes.bodyCopyClass} {...props} />
          </div>
        ) : (
          <div>
            <Headline classes={themeData.classes.headlineText2} {...props} />
          </div>
        )}
        <div>
          <Script
            onLoad={() => {
              console.log('script loaded');
            }}
            src={`https://edge.curalate.com/sites/${siteName}/site/latest/site.min.js`}
          ></Script>
          <div
            className={
              slideFormat === 'inline'
                ? themeData.classes.carouselInline
                : themeData.classes.carouselStack
            }
            data-crl8-container-id={`${containerId}`}
            data-crl8-filter={`${genericFilter}`}
          ></div>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<SocialImageCarouselProps>(SocialImageCarousel);
