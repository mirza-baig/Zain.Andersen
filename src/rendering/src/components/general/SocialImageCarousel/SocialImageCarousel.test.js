// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import SocialImageCarousel from './SocialImageCarousel';
import { inlineData } from './SocialImageCarousel.mock-data';

it('renders correctly', async () => {
  const component = snapshot(SocialImageCarousel, { componentProps: inlineData });
  await hasDataComponent(component, 'general/socialimagecarousel');
});
