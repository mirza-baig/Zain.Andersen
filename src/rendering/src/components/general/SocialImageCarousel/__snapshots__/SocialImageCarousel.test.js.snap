// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/socialimagecarousel"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="flex flex-col md:flex-row col-span-2 md:col-span-6 items-center theme-white"
      >
        <div
          class="p-3 max-w-[315px] md:min-w-[350px]"
        >
          <div
            class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-m md:text-m font-bold mb-s"
          >
            <h2>
              Social Image Carousel Inline
            </h2>
          </div>
          <div
            class=" text-body text-dark-gray"
          >
            <div
              class="body-copy"
            >
              <div>
                This body text only shows up if there are 2 slides selected.
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="w-[375px] md:w-[275px] mmd:w-[403px] ml:w-[611px] mml:w-[627px] lg:w-[845px]"
            data-crl8-container-id="homepage"
            data-crl8-filter="productId:ES-CAS"
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
